Automattic/woocommerce-payments
# Split UPE with a single intent, intent only updated on gateway focus

Fixes #4672

#### Changes proposed in this Pull Request



# Split UPE: Single Intent, No Preloading


This PR implements multiple UPE payment elements working together on a single checkout simultaneously, while sharing a single payment intent. Instead of loading all of the payment elements together on page load, this branch only mounts a payment element (updating the current payment method type on the shared intent) when the gateway is selected for the first time.

## Benefits
 - Less activity on the intent history.
 - Fewer API requests to update intent (probably).

## Costs
 - Since payment elements are not preloaded/mounted on page load, user may have to wait briefly for requests to be made when they change payment gateways.




#### Testing instructions


1. Pull the changes in this PR.
2. Enable the UPE via the WCPay Dev Tools.
3. Enable multiple payment methods from the WCPay Settings and ensure compatible currency is in use.
4. Add item to cart and proceed through a checkout using UPE payment method.
5. Great success.


_BTW, I merged `develop` into this branch, which looks like it hadn't been done in a while, so that's why the commit history is so congested._

-------------------
_I have completed none of the following tasks..._

**Post merge**
